src/target/riscv/riscv-013.c

```diff
@@ -555,9 +567,9 @@ static dmi_status_t dmi_scan(struct target *target, uint32_t *address_in,
 		jtag_add_dr_scan(target->tap, 1, &field, TAP_IDLE);
 	}
 
-	int idle_count = info->dmi_busy_delay;
-	if (exec)
-		idle_count += info->ac_busy_delay;
+	int idle_count = exec
+		? learned_delay(target, RISCV_DELAY_ABSTRACT_COMMAND)
+		: learned_delay(target, RISCV_DELAY_BASE);
```

Comment:
Please, consider dropping the function `learned_delay()`. 

In all functions where it is currently used, it can be simply replaced by calling `riscv_scan_get_delay(info, ...)` and the pointer to `riscv013_info_t` is available in all places where learned_delay() is currently used.

For the sake of code readability, it is IMO better to avoid having multiple functions that do (almost) the same thing. 

```suggestion
	int idle_count = exec
		? riscv_scan_get_delay(info, RISCV_DELAY_ABSTRACT_COMMAND)
		: riscv_scan_get_delay(info, RISCV_DELAY_BASE);
```

Reply:
Addressed.